### PR TITLE
add client size close for #712

### LINKED ISSUE #712: Add the "Connection: close" header to the request, but Stuck at fetching response body
I Add the "Connection: close" header to the request, but my server(opentsdb) don't close connection active, so client stucks when calling `body(response)`. 

The rfc2616 also don't demand that server must close connection active when request with header "Connection: close".
[http://www.ietf.org/rfc/rfc2616.txt](url)
_8.1.2.1 Negotiation

   An HTTP/1.1 server MAY assume that a HTTP/1.1 client intends to
   maintain a persistent connection unless a Connection header including
   the connection-token "close" was sent in the request. **If the server
   chooses to close the connection immediately after sending the
   response, it SHOULD send a Connection header including the
   connection-token close**._

Can we add active close for http client when response finished?

### PR DESCRIPTION
**Pull request by chenzhaoyu**
I attempt to fix #712 , if response body is completed, client side will close connection. 

Comments are welcome!

regards,
chenzhaoyu

### CODE REVIEW

File: boost/network/protocol/http/client/connection/async_normal.hpp [hunk 1]
```diff
@@ -363,14 +363,25 @@ struct http_async_connection
             // Here we handle the body data ourself and append to an
             // ever-growing string buffer.
             auto self = this->shared_from_this();
-            this->parse_body(
+            typename protocol_base::buffer_type::const_iterator begin =
+                this->part_begin;
+            typename protocol_base::buffer_type::const_iterator end = begin;
+            std::advance(end, remainder);
+            string_type body_string(begin, end);
```
Comment (deanberris) [inline boost/network/protocol/http/client/connection/async_normal.hpp]: Do you even need this copy of the string? Why not use the range instead?

File: boost/network/protocol/http/client/connection/async_normal.hpp [hunk 2]
```diff
@@ -476,6 +495,50 @@ struct http_async_connection
     }
   }
 
+  inline bool check_parse_body_complete(string_type& body_string)
```
Comment (deanberris) [inline boost/network/protocol/http/client/connection/async_normal.hpp]: It looks like you could avoid using strings here, and just deal with ranges of iterators.

**Comment by chenzhaoyu** [comment]
@deanberris , thanks for your comments, I have move body check logic to async_protocol_handler.hpp and no extra body buffer be copied now, please check, thanks~

**Comment by chenzhaoyu** [comment]
Hi, @deanberris , can you merge those commits from master into 0.13-release branch?

**Pull request merged**